### DISCUSSION TITLE
custom open command to open md files

### DISCUSSION THREAD

**Discussion by techeddie** [gokcehan/lf#1263] [Q&A] [answered]
Hi, I'm trying to set a default program to open md file extensions. 
Below you can see that I'm using nvim to open files, but I would like to add "glow -p" for md files:

``` 
cmd open ${{
    test -L $f && f=$(readlink -f $f)
    case $(file --mime-type $f -b) in
        text/*) nvim $fx;;     
        *) for f in $fx; do xdg-open $f > /dev/null 2> /dev/null & done;;
    esac
}}
```
I guess I can't use mime type since it shows me text/plain:

```
file --mime-type CONTRIBUTING.md
CONTRIBUTING.md: text/plain
```
Any ideas how I can archive this?

Thanks

**Comment by joelim-work** [comment, marked as answer]
You can just add another `case` statement:

```
cmd open ${{
    test -L $f && f=$(readlink -f $f)
    case $f in
        *.md) glow -p $fx;;  # handle markdown files
        *)
            case $(file --mime-type $f -b) in
                text/*) nvim $fx;;
                *) for f in $fx; do xdg-open $f > /dev/null 2> /dev/null & done;;
            esac
            ;;
    esac
}}
```

You can use this (and `if` statements) to adjust the logic to suit your needs - it's a lot like other programming languages.

**Reply by techeddie** [reply]
beautiful, that work. thanks
